### DISCUSSION TITLE
Is there any plan to add Vue Composition API

### DISCUSSION THREAD

**Discussion by unicape** [wevm/wagmi#1887] [Ideas]
Is there any plan to add Vue Composition API

**Comment by tmm** [comment]
Not by the core team at the moment. If someone wants to work on this, happy to chat. (There's a SolidJS adapter in-progress right now #1677.)

**Reply by unicape** [reply]
that's great i'd love to oblige

**Reply by Hebilicious** [reply]
There's already a working vue version of wagmi/react : [vagmi](https://github.com/wobsoriano/vagmi). Would you be interested in a PR with most of the work moved here ? It would be nice to consolidate everything under this monorepo.

**Reply by unicape** [reply]
I know this but it's slow to update
[use-wagmi](https://github.com/unicape/use-wagmi) Already supports most APIs and is developing for wagmi@1.0

**Reply by Hebilicious** [reply]
I didn't know you had rolled your own @unicape !
Now that wagmi v1 is out, looks like it's the perfect time to open a PR with an "official" vue integration ? @wobsoriano (vagmi dev) was open to the idea https://github.com/wobsoriano/vagmi/issues/48

**Reply by unicape** [reply]
@Hebilicious You're right I'll try to raise a PR when I'm done developing V1
At present, it is developed as an independent package because it is relatively free and convenient

**Reply by unicape** [reply]
@tmm @jxom If it is inherited to the official release, what requirements are required, such as unit test coverage? If it is not completed, can the internal test version be released?

**Reply by percyzt** [reply]
> 这个我知道 但是更新很慢 [use-wagmi](https://github.com/unicape/use-wagmi)已经支持大部分API，正在为wagmi@1.0开发

Does it support transactions and network switching?

**Reply by wobsoriano** [reply]
Hey @unicape, if you have more time to maintaining that Vue package, I can retire the vagmi repo and point them to that repo in the README 😄  Tell me if you're interested in that idea 😃
